h5py/h5d.pyx

```diff
@@ -430,3 +431,65 @@ cdef class DatasetID(ObjectID):
                 efree(offset)
                 if space_id:
                     H5Sclose(space_id)
+
+    IF HDF5_VERSION >= (1, 10, 2):
+
+        def read_direct_chunk(self, offsets, filter_mask=0xFFFF, PropID dxpl=None):
+            """ (offsets, H5Z_filter_t filter_mask=[], PropID dxpl=None)
+
+            Reads data to a bytes array directly from a chunk at position
+            specified by the offsets argument.
+
+            Returns a tuple containing the bytes data, and the filter_mask which
+            was used for this data.
+
+            Feature requires: 1.10.2 HDF5
+            """
+
+            cdef hid_t dset_id
+            cdef hid_t dxpl_id
+            cdef hid_t space_id = 0
+            cdef hsize_t *offset = NULL
+            cdef size_t data_size
+            cdef int rank
+            cdef uint32_t filters = 0
+            cdef hsize_t read_chunk_nbytes
+            cdef array.array data = array.array('B')
+            cdef herr_t status
+
+            if filter_mask is None:
+                # Skip all the filters
+                filters = 0xFFFF
+            else:
+                filters = int(filter_mask)
+
+            dset_id = self.id
+            dxpl_id = pdefault(dxpl)
+            space_id = H5Dget_space(self.id)
+            rank = H5Sget_simple_extent_ndims(space_id)
+
+            if offsets is not None:
+                if len(offsets) != rank:
+                    raise TypeError("offset length (%d) must match dataset rank (%d)" % (len(offsets), rank))
+
+            try:
+                offset = <hsize_t*>emalloc(sizeof(hsize_t)*rank)
+                convert_tuple(offsets, offset, rank)
+                status = H5Dget_chunk_storage_size(dset_id, offset, &read_chunk_nbytes)
+                if status < 0:
```

Comment:
I don't think you need the status checks; h5py should autogenerate a function in `defs.pyx` which wraps the HDF5 function to turn errors into Python exceptions.